tests/test_key.py

```diff
@@ -52,20 +52,23 @@ def test_custom_getprime_func(self):
         # List of primes to test with, in order [p, q, p, q, ....]
         # By starting with two of the same primes, we test that this is
         # properly rejected.
-        primes = [64123, 64123, 64123, 50957, 39317, 33107]
+        primes = [64123, 64123,#discarded because identical
```

Comment:
This unit test mostly worked with the new keygen logic but still needed a few minor changes.
The final accepted primes were changed along with the `nbits` argument to `gen_keys`. This prevents an modulo bit length check from tripping in the new keygen logic.